File: NetFramework/WinForms/CS/ControlExplorer/RulesManager/Dialogs/SingleValueConditionalFormattingForm.cs
using ControlExplorer.RulesManager.Presets;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ControlExplorer.RulesManager.Dialogs
{
    public partial class SingleValueConditionalFormattingForm : BaseConditionalFormattingForm
    {
        public SingleValueConditionalFormattingForm()
        {
            InitializeComponent();
        }

        private void SingleValueConditionalFormattingForm_Load(object sender, EventArgs e)
        {
            cmbStylePreset.SelectedIndex = 0;
        }

        [Browsable(false)]
        public Highlight SelectedStylePreset
        {
            get
            {
                var selectedStylePresetText = cmbStylePreset.Text;
                var stylePresets = Enum.GetValues(typeof(Highlight)).Cast<Highlight>().ToList();

                return stylePresets.First(stylePreset => stylePreset.ToString().ToLower() == selectedStylePresetText.ToLower().Replace(" ", ""));
            }
        }

        [Browsable(false)]
        public string Value
        {
            get
            {
                var textValue = tbValue?.Text;
                if (textValue != null)
                {
                    return textValue;
                }

                return string.Empty;
            }
        }
    }
}

File: NetFramework/RulesManager/VB/ExcelLikeConditionalFormatting/Dialogs/SingleValueConditionalFormattingForm.vb
Imports System.ComponentModel

Public Class SingleValueConditionalFormattingForm
    Private Sub SingleValueConditionalFormattingForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        cmbStylePreset.SelectedIndex = 0
    End Sub

    <Browsable(False)>
    Public ReadOnly Property SelectedStylePreset As Highlight
        Get
            Dim selectedStylePresetText = cmbStylePreset.Text
            Dim stylePresets = [Enum].GetValues(GetType(Highlight)).Cast(Of Highlight)().ToList()

            Return stylePresets.First(Function(stylePreset) stylePreset.ToString().ToLower() = selectedStylePresetText.ToLower().Replace(" ", ""))
        End Get
    End Property

    <Browsable(False)>
    Public ReadOnly Property Value As String
        Get
            Dim textValue = tbValue?.Text

            If textValue IsNot Nothing Then
                Return textValue
            End If

            Return String.Empty
        End Get
    End Property

End Class